ejb3/src/main/java/org/jboss/as/ejb3/tx/BMTInterceptor.java

```diff
@@ -60,7 +59,7 @@ public Object processInvocation(final InterceptorContext context) throws Excepti
                 if (oldTx != null) tm.resume(oldTx);
             }
         } finally {
-            tm.setTransactionTimeout(oldTimeout == -1 ? 0 : oldTimeout);
+            tm.setTransactionTimeout(oldTimeout == ContextTransactionManager.getGlobalDefaultTransactionTimeout() ? 0 : oldTimeout);
```

Comment:
The WFLY-10146 change included the following comment before this line, which I'd like to see here as well as it helps clarify what's otherwise pretty confusing:

// See also https://issues.jboss.org/browse/WFTC-44

The part of the WFLY-10146 change adding that comment:

https://github.com/wildfly/wildfly/pull/11086/files#diff-e05009bda935bba48b4ee7ca633ba1ecR170

The rest of the changes in this class all LGTM; they are just changing to standard ContextTransactionManager use. I think this line seems fine too (other than needing the comment) but checking with David or @chengfang or @tadamski makes sense.

Reply:
I agree with Brian, adding the comment would be a good idea.  Otherwise IIRC this seems correct.

Reply:
@ochaloup Please re-add the comment :)

Reply:
@kabir just I'm now not sure what you are talking about. I added the comment `// See also https://issues.jboss.org/browse/WFTC-44` above the change of the `tm.setTransactionTimeout(...)`. I can see the change is still at the place if I take a look to the PR "Files changes". Could be that only the github does not show this as "outdated" as I did not do any change on the highlighted line but above it.

Reply:
@ochaloup sorry, you are right